src/components/fx/helpers.js

```diff
@@ -11,10 +11,30 @@
 var Lib = require('../../lib');
 
 // look for either subplot or xaxis and yaxis attributes
+// does not handle splom case
 exports.getSubplot = function getSubplot(trace) {
     return trace.subplot || (trace.xaxis + trace.yaxis) || trace.geo;
 };
 
+// is trace in given list of subplots?
+// does handle splom case
+exports.isTraceInSubplots = function isTraceInSubplot(trace, subplots) {
```

Comment:
:pray:

Reply:
Not sure how this will work with `show(upper|lower)half` - actually, it's probably better to let it pass here - in fact in case there's ever another trace type that spans multiple subplots, we could do `if(trace.xaxes && trace.yaxes)` instead of `if(trace.type === 'splom')` - and then bail out during `hoverPoints` for the trace if that subplot is blank.

Reply:
> Not sure how this will work with show(upper|lower)half

Turns out this works fine for splom traces with `show(upper|lower)half` and `diagonal.visible: false` as hovering over their omitted subplots never triggers `Fx.hover` as we don't draw those subplots and more precisely we don't draw their corresponding `<g .draglayer>`.

If we choose at some point to draw only one drag layer for all interaction types for all subplots, we'll of course have to change this logic here. But if ok, I'll leave this out of this PR and list it in an upcoming _splom open items_ issue.